### ISSUE TITLE
npm run dev Issue

### ISSUE THREAD

**Issue by Skorpion7777** [blockstack/blockstack-browser#299]
**System:** plain Virtual Box Ubuntu installation. With Ubuntu 16.04.2 LTS
Shure I've installed all necessary packages (newest release): git, pip, npm, virtualenv, node.js

After that I've followed @larrysalibra instructions:

```
apt-get update && sudo apt-get install -y python-pip python-dev libssl-dev libffi-dev rng-tools libgmp3-dev

virtualenv -p /usr/bin/python2.7 blockstack

source blockstack/bin/activate

pip install git+https://github.com/blockstack/virtualchain.git@rc-0.14.1 git+https://github.com/blockstack/blockstack-profiles-py.git@rc-0.14.1 git+https://github.com/blockstack/dns-zone-file-py.git@rc-0.14.1 

pip install git+https://github.com/blockstack/blockstack-core.git@089d436caea79919296fbc05d6d46e5e52b57d6c
```

That worked fine and after that`blockstack setup` and `blockstack api start` also worked fine. And localhost showed some Infos and no Error. So Step 1 was completed.

After that:

> (blockstack) test@test-VirtualBox:~/blockstack-browser$ npm install

worked fine. After that:

> (blockstack) test@test-VirtualBox:~/blockstack-browser$ npm run dev

but this step crashes and i get the error Code:

```
npm ERR! Linux 4.4.0-31-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "dev"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE
npm ERR! blockstack-portal@0.4.0 dev: `concurrently --kill-others "npm run dev-ui" "npm run dev-proxy"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the blockstack-portal@0.4.0 dev script 'concurrently --kill-others "npm run dev-ui" "npm run dev-proxy"'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the blockstack-portal package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     concurrently --kill-others "npm run dev-ui" "npm run dev-proxy"
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs blockstack-portal
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls blockstack-portal
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/test/blockstack-browser/npm-debug.log

```

Thanks for your help!

**Comment by shea256** [comment]
@Skorpion7777 Can you try upgrading to node 6 or node 7?

**Comment by Skorpion7777** [comment]
@shea256  I've now upgraded to node v7.7.2 but I'm still receiving the same errror when using `npm run dev` in ~/blockstack-browser

**Comment by shea256** [comment]
OK can you please try the following:

1. Switch to node 6
1. Switch to npm 4
1. Make sure you've run `npm install`
1. Run `npm run dev-ui` in one console tab
1. Run `npm run dev-proxy` in another console tab

**Comment by Skorpion7777** [comment]
**Update:** since I'had issues with npm, nodejs install, reinstall etc. so everything got messed up and I stopped and opend a new maschine. Same thing. Step 1 as written above everything fine and localhost shows:

> {"status": "registered", "zonefile": "$ORIGIN blockstack.id\n$TTL 3600\n_http._tcp IN URI 10 1 \"https://blockstack.s3.amazonaws.com/blockstack.id\"\n", "last_txid": "a3ad2dd88ea174c185ef8a07fc24b7bd4bfff09878a13545747a08326b1e8e24", "zonefile_hash": "36594b8e985f6edd3499863b4e62d217f6567c67", "address": "1Ed1qATBtiD8CAxovZZZuStHV39gothu7E"}

Now my commands for step 2 (right order):
`git clone https://github.com/blockstack/blockstack-browser.git`
`curl -sL https://deb.nodesource.com/setup_6.x -o nodesource_setup.sh`
`vim nodesource_setup.sh`
`sudo apt-get install vim-nox`
`vim nodesource_setup.sh`
`sudo bash nodesource_setup.sh`
`sudo apt-get install nodejs`
`sudo apt-get install build-essential libssl-dev`
`sudo npm install npm@latest -g`
`cd blockstack-browser/`
`npm install`

went fine 2 warnings for: "SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0" and "SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents" but both optional

` npm run dev` was the last command and I got:

>   ----------------------------------
> [0]        Local: http://localhost:3000
> [0]     External: http://10.0.2.15:3000
> [0]  ----------------------------------
> [0]           UI: http://localhost:3001
> [0]  UI External: http://10.0.2.15:3001

nice one. Now I'm in the Main Menu of the Blockstack Browser but still if I chose "Hello, Blockstack" or want to register a new .id in Profiles it Errors which means I'm not connected to the Blockstack web which is weird since blockstack core seems to run.

PS: **Versions:** : npm: 4.4.1 ; nodejs: v6.10.0

Thanks for your help so far!

**Comment by larrysalibra** [comment]
@Skorpion7777 Thanks so much for the very detailed response and giving it another go! 

You'll need to make sure that your Blockstack Core API endpoint is running at the location the Portal expects. By default this *should* be localhost:6270. You can check the Portal API settings in the Account app and also check the network requests box of the developers tools to see where requests are going and if they're being answered by Core.

**Comment by Skorpion7777** [comment]
@larrysalibra I've checked my `blockstack setup` window (I still had it opened) and everything was set to default. So the api_endpoint is 6270.
When I go to localhost:6270 I only get a blank page. The firefox dev Tool shows: 404 Not Found. Thats the Request Headers:

> Host: localhost:6270
> User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:47.0) Gecko/20100101 Firefox/47.0
> Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
> Accept-Language: en-US,en;q=0.5
> Accept-Encoding: gzip, deflate
> Cookie: io=SsSebVivxjxDgJjXAAAJ; mp_mixpanel__c=2; mp_f6808747c935f2b51cf0563b5d8cbd04_mixpanel=%7B%22distinct_id%22%3A%20%2215ab88269c876-0d63b609bd3674-72226750-1bf300-15ab88269c920c%22%2C%22%24initial_referrer%22%3A%20%22https%3A%2F%2Fwww.dropbox.com%22%2C%22%24initial_referring_domain%22%3A%20%22www.dropbox.com%22%7D
> Connection: keep-alive
> Cache-Control: max-age=0

When I check with the dev tool at the name registration I get 2 Errors 404:

first at (GET) (...).id    localhost:6270   json with:

> Host: localhost:6270
> User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:47.0) Gecko/20100101 Firefox/47.0
> Accept: */*
> Accept-Language: en-US,en;q=0.5
> Accept-Encoding: gzip, deflate
> origin: http://localhost:3000
> Referer: http://localhost:3000/profiles/register
> Connection: keep-alive

and one for (POST) "delete"   with the domain: "api.dropboxapi.com"  json

> Host: api.dropboxapi.com
> User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:47.0) Gecko/20100101 Firefox/47.0
> Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
> Accept-Language: en-US,en;q=0.5
> Accept-Encoding: gzip, deflate, br
> Content-Type: application/json
> Authorization: Bearer vFvL75PYnYoAAAAAAAADNE1Yuo4inWROdUjcAQ3H8K4LjA0kJohAElnDqWWxpdyv
> Referer: http://localhost:3000/profiles/register
> Content-Length: 38
> origin: http://localhost:3000
> Connection: keep-alive

with the request Body: 

> {"path":"/ertetertet.id/profile.json"}

Two are Error 500 Internal server Errors:

First at (GET) (...).id  over localhost:6270

> Host: localhost:6270
> User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:47.0) Gecko/20100101 Firefox/47.0
> Accept: */*
> Accept-Language: en-US,en;q=0.5
> Accept-Encoding: gzip, deflate
> origin: http://localhost:3000
> Referer: http://localhost:3000/profiles/register
> Connection: keep-alive

Body: 
> {"name":"ertetertet.id","owner_address":"16EBjTP9KEipeqEeRSQgD4asKbwyyskkGX","zonefile":"$ORIGIN ertetertet.id\n$TTL 3600\n_http._tcp\tIN\tURI\t10\t1\t\"https://www.dropbox.com/s/39zn1etltxv9uar/profile.json?dl=1\"\n\n","min_confs":0}

and the second is (POST) names  at localhost:6270

> Host: localhost:6270
> User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:47.0) Gecko/20100101 Firefox/47.0
> Accept: application/json
> Accept-Language: en-US,en;q=0.5
> Accept-Encoding: gzip, deflate
> Content-Type: application/json
> origin: http://localhost:3000
> Referer: http://localhost:3000/profiles/register
> Content-Length: 234
> Connection: keep-alive

Body:

> {"name":"ertetertet.id","owner_address":"16EBjTP9KEipeqEeRSQgD4asKbwyyskkGX","zonefile":"$ORIGIN ertetertet.id\n$TTL 3600\n_http._tcp\tIN\tURI\t10\t1\t\"https://www.dropbox.com/s/39zn1etltxv9uar/profile.json?dl=1\"\n\n","min_confs":0}


I'm sorry for the long post but maybe it'll solve the Problem

**Comment by larrysalibra** [comment]
@Skorpion7777 Have you sent money to your bitcoin wallet? (In the wallet app) You might be running into a bug that exists in that version of core where the API end point crashes if your bitcoin wallet has no bitcoin. This is fixed in the most recent version of core, however we haven't yet ported Portal over to the updated API.

See if you can browse profiles: 

Go to Profiles and enter `ryan.id` (or click here: http://localhost:3000/profiles/blockchain/ryan.id). You should see @shea256 's profile.

You can also check the `api_endpoint.log` file in your Core configuration directory (which defaults to `~/.blockstack/`) to see what is happening when the API is called.

---

Another thing: the credentials for your Dropbox are in your previous comment: (the `Authorization: Bearer` header). Someone could take that it use it to access your the Blockstack app folder of your Dropbox. You should probably disconnect Blockstack both *in Dropbox* and the Portal. Once you've done that, you can reconnect it in the Portal. You can disconnect Blockstack (and revoke the exposed credentials) from the App section of this page: https://www.dropbox.com/account/security

Thanks again for taking the time to go through this. Sorry again about the problem problems you've encountered! Really appreciate your bug reports!!

**Comment by Skorpion7777** [comment]
@larrysalibra no it's a completly new wallet, so no Bitcoins. You've been right. I can see Ryans Profile. If you think putting some satoshis into the account will solve the problem I'll do so on monday. Thanks for the Dropbox warning, I fixed it! I can send you the api_endpoint.log via a PM in slack if you're interested. but for today I'm pretty much done. As I said if you think some BTC could solve the problem I'll do so on monday.
By the way, have you ever thought about providing a docker image for a simpler installation? That would be awesome.

And thanks again for your help! Response speed is over the top!

**Comment by larrysalibra** [comment]
@Skorpion7777 I think @muneeb-ali and another community member may be working docker images for Blockstack Core. 

On macOS, we've created a native app that bundles Core and Portal together so that users don't have to go through this hassle. I'd love to start a discussion about the best way to distribute Blockstack (Portal + Core) on Linux for end users. I just started a thread on the forum to discuss this and sent you an invite. Would love your thoughts! https://forum.blockstack.org/t/distributing-blockstack-for-end-users-on-linux/809?u=larry
